internal/controller/kustomization_controller.go

```diff
@@ -965,6 +972,11 @@ func finalizerShouldDeleteResources(obj *kustomizev1.Kustomization) bool {
 
 func (r *KustomizationReconciler) finalize(ctx context.Context,
 	obj *kustomizev1.Kustomization) (ctrl.Result, error) {
+
+	// Here the origin revision is not known because the artifact was not fetched,
+	// it's too soon in the Reconcile() method.
+	const originRevision = ""
```

Comment:
Does this sound right?

Reply:
If we add `status.LastAppliedOriginRevision` than we can just read it if it's there, like we do for revision now.

Reply:
fixed, thanks